namui/src/namui/common/types/macros.rs

```diff
@@ -1,21 +1,34 @@
 #[macro_export]
+/// Please add serde into your Cargo.toml
+/// ```toml
+/// serde = { version = "1.0", features = ["derive"] }
+/// ```
 macro_rules! common_for_f32_type {
     ($your_type: tt, $short_term: ident, $short_term_ext: ident) => {
-        $crate::types::macros::common_for_f32_type!($your_type, |lhs: $your_type| -> f32 {
+        $crate::types::common_for_f32_type!($your_type, |lhs: $your_type| -> f32 {
             lhs.0
         }, |rhs: f32| -> $your_type {
             $your_type(rhs)
-        }, $short_term, $short_term_ext);
+        }, $short_term, $short_term_ext, not_ratio);
+    };
+    ($your_type: tt, $to_f32: expr, $from: expr, $short_term: ident, $short_term_ext: ident, ratio) => {
+        $crate::types::common_for_f32_type!($your_type, $to_f32, $from, $short_term, $short_term_ext);
+    };
+    ($your_type: tt, $to_f32: expr, $from: expr, $short_term: ident, $short_term_ext: ident, not_ratio) => {
+        $crate::types::common_for_f32_type!($your_type, $to_f32, $from, $short_term, $short_term_ext);
+        $crate::types::impl_op_forward_ref!(/|x: $your_type, y: $your_type| -> f32 {
+            x.0 / y.0
+        });
```

Comment:
No need for Mul(*) right?

Reply:
Yes it's only for same dimension division.